dhis-2/dhis-api/src/main/java/org/hisp/dhis/merge/MergeService.java

```diff
@@ -39,6 +40,27 @@
  */
 public interface MergeService {
 
+  /**
+   * Processes a merge in full.
+   *
+   * @param mergeParams {@link MergeParams} to process
+   * @return updated {@link MergeReport} with any errors
+   */
+  @Transactional
+  default MergeReport processMerge(@Nonnull MergeParams mergeParams) throws ConflictException {
+    MergeReport mergeReport = new MergeReport();
+
+    MergeRequest mergeRequest = validate(mergeParams, mergeReport);
+    if (mergeReport.hasErrorMessages())
+      throw new ConflictException("Merge validation error").setMergeReport(mergeReport);
+
+    MergeReport report = merge(mergeRequest, mergeReport);
+    if (report.hasErrorMessages())
+      throw new ConflictException("Merge error").setMergeReport(mergeReport);
+
```

Comment:
There are two variables `mergeReport` and `report`. Seems there should only be one ?

Reply:
yes, essentially the same, but can get rid of one to keep it clearer.
Nice spot.